metrics/prometheus/factory.go

```diff
@@ -184,7 +184,7 @@ func (f *Factory) subScope(name string) string {
 	if name == "" {
 		return f.normalize(f.scope)
 	}
-	return f.normalize(f.scope + ":" + name)
+	return f.normalize(f.scope + "_" + name)
```

Comment:
@jpkrohling we had an internal discussion, the proposal is that instead of hardcoding this separator we make it an option to the factory (since it already supports options). Then in jaeger we default to `_` (without bothering to make it a cli switch), but in jaeger-client-go people can, if necessary, instantiate the factory with `:` as before.

Reply:
I just updated the PR to include this suggestion. I decided to use a `rune` for the separator, as I think a single char should suffice, but let me know if a string would be more suitable.